Tests/Marketplace/search_and_install_packs.py

```diff
@@ -675,33 +662,34 @@ def search_and_install_packs_and_their_dependencies(pack_ids: list,
 
     logging.info(f'Starting to search and install packs in server: {host}')
 
-    packs_to_install: list = []  # we save all the packs we want to install, to avoid duplications
-    installation_request_body: list = []  # the packs to install, in the request format
-    batch_packs_install_request_body: list = []    # list of lists of packs to install if install packs one by one.
+    packs_to_install: list = []  # Packs we want to install, to avoid duplications
+    installation_request_body: list = []  # Packs to install, in the request format
+    batch_packs_install_request_body: list = []  # List of lists of packs to install if not using multithreading .
     # Each list contain one pack and its dependencies.
-    packs_in_the_list_to_install: list = []    # list of packs that are already in the list to install.
+    collected_dependencies: list = []  # List of packs that are already in the list to install.
+    is_post_update = os.getenv('BUCKET_UPLOAD', 'false').casefold() == 'true'
```

Comment:
This function (`search_pack_and_its_dependencies`) is called both on pre-update, and on post-update.(BUCKET_UPLOAD` is set to `True` when running on post-update. It's a pre-existing env variable that I'm using.

The issue is, that at first, I just used the marketplace API for checking whether a pack is deprecated or not, but then we found that this caused issues, since if we deprecate a pack in the current PR, the API call will fail (pack wasn't uploaded to the bucket, so it returns a message saying the requested pack ID doesn't exist). Like how for example, the Microsoft Defender pack which has a test that deprecated it, failed the build because the API call on post-update failed, since this pack wasn't in the marketplace.

So the solution was, that if we're running on pre-update, we check for deprecation status using the marketplace. If we're running on post-update, we'll check deprecation status locally by checking the file (this way we will see the deprecation status according to the changes on the PR.)

Reply:
After a meeting, we have concluded that checking for whether we're in post-update mode using the `BUCKET_UPLOAD` env var is invalid, since it doesn't cover all cases.

I'm working on a fix for that.

Reply:
Fixed on 35dd44b0372cfe14040ba1832120765be56ceba0